src/plot.js

```diff
@@ -97,6 +97,18 @@ export function plot(options = {}) {
         ? facetsExclude || (facetsExclude = facetsIndex.map((f) => Uint32Array.from(difference(facetIndex, f))))
         : undefined;
     const {data, facets, channels} = mark.initialize(markFacets, facetChannels);
+    if (
+      facets && // if truthy, faceting is used
+      mark.facet === "auto" && // if the user didn't specify a facet
+      mark.data &&
+      (mark.data.length === facetIndex?.length || mark.data.size === facetIndex?.length) && // for Maps
+      mark.data !== facet?.data &&
+      facetIndex?.length > 1
```

Comment:
Based on previous comments, here’s my take on the logic:

```suggestion
      facets && // faceting is present
      facetIndex.length > 1 && // more than one facet (non-trivial faceting)
      mark.facet === "auto" && // no explicit mark facet option
      mark.data !== facet.data && // mark is implicitly not faceted (different data)
      arrayify(mark.data)?.length === facetIndex.length // mark data seems parallel to facet data
```